lib/AppInfo/Application.php

```diff
@@ -78,4 +81,33 @@ public function __construct(array $urlParams = array()) {
 		});
 	}
 
+	public function boot() {
+		$this->getContainer()->query('UserHooks')->register();
+		$request = $this->getContainer()->getServer()->getRequest();
+		$redirectUrl = $request->getParam('redirect_url');
+		if ($redirectUrl === null) {
+			return;
+		}
+
+		$urlParts = parse_url(urldecode($redirectUrl));
+		if (strpos($urlParts['path'], 'apps/oauth2/authorize') === false) {
+			return;
+		}
+		$params = [];
+		parse_str($urlParts['query'], $params);
+		if (!isset($params['client_id'])) {
```

Comment:
You are checking for `redirect_uri` and `client_id` but [in the controller function](https://github.com/owncloud/oauth2/blob/master/lib/Controller/PageController.php#L103-L104) there are some more checks.